### PR TITLE
Consent updates part 2

### PR DESCRIPTION
**Pull request by bytheway875**
Consent updates part 2 - create an additional consent pdf, a consents table, some factories, and consolidate logic for pdf creation through intake.

### CODE REVIEW

File: spec/factories/consents.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#  index_consents_on_client_id  (client_id)
+#
+FactoryBot.define do
+  factory :consent do
+    disclose_consented_at { DateTime.current }
+    use_consented_at {  DateTime.current }
+    global_carryforward_consented_at { DateTime.current }
+    relational_efin_consented_at { DateTime.current }
+    ip { IPAddr.new }
+    user_agent { "CERN-NextStep-WorldWideWeb.app/1.1 libwww/2.07" }
```
Comment (alaroia-cfa) [inline spec/factories/consents.rb:27]: ❤️

File: spec/models/intake_spec.rb [hunk 2]
```diff
@@ -1049,38 +1050,71 @@
...
+    before do
+      example_pdf = Tempfile.new("example.pdf")
+      example_pdf.write("example pdf contents")
+      allow(intake).to receive(:pdf).and_return(example_pdf)
+    end
 
-  describe "#update_or_create_14446_document" do
+    let(:intake) { create(:intake) }
+
+    context "when there is not an existing 13614-C document" do
```
Comment (alaroia-cfa) [inline spec/models/intake_spec.rb:1062]: This test is not strictly doc-type-specific, so the context could be renamed `"when there is no doc of the given type for the client"`. Not a blocker.

**Review by alaroia-cfa — APPROVED** [review]
Splendid. Simple and good.

File: app/lib/additional_consent_pdf.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class AdditionalConsentPdf
```
Comment (jenny-heath) [inline app/lib/additional_consent_pdf.rb:1]: maybe update name to "OptionalConsentPdf"

File: spec/models/intake_spec.rb [hunk 4]
```diff
@@ -1049,38 +1050,71 @@
...
     before do
       example_pdf = Tempfile.new("example.pdf")
       example_pdf.write("example pdf contents")
-      allow(ConsentPdf).to receive(:new).and_return(double(output_file: example_pdf))
+      allow(AdditionalConsentPdf).to receive(:new).and_return(double(output_file: example_pdf))
     end
 
-    context "when there is not an existing 14446 document" do
+    context "when there is not an existing additional consent document" do
       it "creates a 14446 PDF with a given filename" do
```
Comment (jenny-heath) [inline spec/models/intake_spec.rb:1097]: wrong description

**Pull request merged**